lib/datasets/dataset.rb

```diff
@@ -33,9 +33,23 @@ def cache_path
       @cache_path ||= CachePath.new(@metadata.id)
     end
 
-    def download(output_path, url, &block)
-      downloader = Downloader.new(url)
-      downloader.download(output_path, &block)
+    def download(output_path, url, *fallback_urls, &block)
+      urls = [url] + fallback_urls
+      urls.each_with_index do |url, idx|
+        downloader = Downloader.new(url)
+        downloader.download(output_path, &block)
+        return
+      rescue Net::HTTPClientException => error
+        if urls.last != url
+          failed_url = "<#{url}>"
+          fallback_url = "<#{urls[idx + 1]}>"
+          message = "#{error.response.code} #{error.response.message}: " +
+                    "fallback: #{failed_url} -> #{fallback_url}"
+          $stderr.puts(message)
+        else
+          raise error
```

Comment:
We should not specify `error` here to propagate the raised exception as-is:

```suggestion
          raise
```

Reply:
Oh, I didn't know. Thanks for your suggestion!

https://docs.ruby-lang.org/en/3.3/Kernel.html#method-i-raise
> With no arguments, raises the exception in `$!` or raises a [RuntimeError](https://docs.ruby-lang.org/en/3.3/RuntimeError.html) if `$!` is `nil`.